src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,92 @@
+import classNames from 'classnames';
 import React from 'react';
+import { addComment } from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Errors } from '../types/Errors';
+import { Post } from '../types/Post';
+
+type Props = {
+  selectedPost: Post | undefined,
+  comments: Comment[] | undefined,
+  setComments(comments: Comment[]): void,
+  setError(error: Errors): void
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  comments,
+  setComments,
+  setError,
+}) => {
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [name, setName] = React.useState('');
+  const [nameError, setNameError] = React.useState(false);
+  const [email, setEmail] = React.useState('');
+  const [emailError, setEmailError] = React.useState(false);
+  const [body, setBody] = React.useState('');
+  const [bodyError, setBodyError] = React.useState(false);
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!name.trim() || !email.trim() || !body.trim()) {
+      if (!name.trim()) {
+        setNameError(true);
+      }
+
+      if (!email.trim()) {
+        setEmailError(true);
+      }
+
+      if (!body.trim()) {
+        setBodyError(true);
+      }
+
+      return;
+    }
+
+    if (selectedPost) {
+      const commentToAdd: Comment = {
+        id: 0,
+        postId: selectedPost.id,
+        name,
+        email,
+        body,
+      };
+
+      setIsLoading(true);
+
+      let newComment: Comment;
+
+      addComment(commentToAdd)
+        .then(result => {
+          newComment = result;
+        })
+        .then(() => {
+          setComments(comments ? [...comments, newComment] : [newComment]);
+        })
```

Comment:
```suggestion
        .then(result => {
          setComments(comments ? [...comments, result] : [result]);
        })
```

don't get why do you need additional `then` here